networking/api.go

```diff
@@ -767,7 +777,8 @@ func RequestOrRenewDhcpLease(ifname string, leaseinfo *DhcpLeaseInfo, opts *dhcp
 	}
 	debugging.DEBUG_OUT("leaseinfo %+v\n", leaseinfo)
 
-	if leaseinfo == nil || !leaseinfo.IsValid() {
+	//If the lease is not valid or expired, get a new lease
+	if leaseinfo == nil || !leaseinfo.IsValid() || !leaseinfo.IsExpired() {
```

Comment:
> !leaseinfo.IsExpired()

Why is the `!` needed here? I would think that if a lease has expired, it wants to request a new lease.

Reply:
I think the standard states that you need to renew before expiration, otherwise you just request a lease (not a renew). @SenRamakri - do you know the story?

Reply:
@cmonr  - it's up to you to approve. I am on the road until Thurs. can't test.

Reply:
@cmonr - You are right, its more logical to sat if lease. expired do a renew. It still works because the IsExpired() function returns false when expired which made my testing succeed, my mistake. I'll modify it to align with the wording.

@edhemphill - Correct me if Im wrong, if the renewal time and rebind time has passed(T1 and T2 as in spec, please see - https://tools.ietf.org/html/rfc2131) should we go ahead request instead of renewing?

Reply:
@cmor - I have fixed the IsExpired() return logic.

Reply:
> It still works because the IsExpired() function returns false when expired

That was surprising. Sounds like a good unit test to add 😁